src/Core/Extensible.php

```diff
@@ -549,21 +556,26 @@ public function getExtensionInstances()
             if (in_array($class, self::$unextendable_classes)) {
                 continue;
             }
-            $extensions = Config::inst()->get($class, 'extensions', Config::UNINHERITED | Config::EXCLUDE_EXTRA_SOURCES);
-
-            if ($extensions) {
-                foreach ($extensions as $extension) {
-                    $name = $extension;
-                    // Allow service names of the form "%$ServiceName"
-                    if (substr($name, 0, 2) == '%$') {
-                        $name = substr($name, 2);
-                    }
-                    $name = trim(strtok($name, '('));
-                    if (class_exists($name)) {
-                        $name = ClassInfo::class_name($name);
-                    }
-                    $this->extension_instances[$name] = Injector::inst()->get($extension);
+            $extensions = Config::inst()->get(
+                $class,
+                'extensions',
+                Config::UNINHERITED | Config::EXCLUDE_EXTRA_SOURCES
+            ) ?? [];
+
+            foreach ($extensions as $extension) {
+                // Allow service names of the form "%$ServiceName" or "ServiceName.default"
```

Comment:
Makes it sound like it's either or, I'd list all 4 combindations like you have in your unit test i.e.
"%$ServiceName" or "%$ServiceName.default" or "ServiceName" or "ServiceName.default"